### PR TITLE
DIV and MOD operations should be unsigned

### LINKED ISSUE #395: DIV instruction should be unsigned division
Per bpf mailing list discussion the ebpf DIV instruction is unsigned division, not signed division.

https://lore.kernel.org/bpf/20220930205211.tb26v4rzhqrgog2h@macbook-pro-4.dhcp.thefacebook.com/ 

> BPF ISA only supports unsigned div/mod.
> There are no instructions for signed div/mod.

### PR DESCRIPTION
**Pull request by dthaler**
Fixes #395

PR #384 was incorrect in that it assumed division and modulo were signed operations.
As issue #395 points out, the eBPF ISA defines them to instead be unsigned operations.
For division with a signed immediate value, the value is first sign extended to 64 bits and then the 64-bit value is reinterpreted as an unsigned value.  (This is being clarified in the eBPF ISA documentation.)

Signed-off-by: Dave Thaler <dthaler@microsoft.com>

### CODE REVIEW

File: test-data/divmod.yaml [hunk 1]
```diff
@@ -392,41 +393,55 @@ post:
...
+  - r2.value=[1, 5]
+---
 test-case: negative modulo positive
 
 pre: ["r1.type=number", "r1.value=-13"]
 
 code:
   <start>: |
-    r1 %= 3
+    r1 %= 4
```
Comment (elazarg) [inline test-data/divmod.yaml:416]: Why the change?

File: test-data/divmod.yaml [hunk 2]
```diff
@@ -392,41 +393,55 @@ post:
...
+  - r1.value=3
 ---
 test-case: positive modulo negative
 
 pre: ["r1.type=number", "r1.value=13"]
 
 code:
   <start>: |
-    r1 %= -3
+    r1 %= -3 ; 13 % 18446744073709551613
```
Comment (elazarg) [inline test-data/divmod.yaml:428]: Maybe we should print the unsigned version for unsigned operation?

**Review by elazarg — APPROVED** [review]
LGTM. Just a quick question.

File: test-data/divmod.yaml (hunk 1 above)
Comment (dthaler) [reply test-data/divmod.yaml:416]: because -13 % 3 is 0 which wouldn't necessarily distinguish the case from other paths that set the answer to 0 so I wanted a non-zero result case

File: test-data/divmod.yaml (hunk 2 above)
Comment (elazarg) [reply test-data/divmod.yaml:428]: In general, change the way we print.

**Comment by elazarg** [comment]
@dthaler I missed that no checks were run on this branch (I thought they all pass), and now they are failing on main. I have no idea why the tests did not run here. Do you?
